homeassistant/components/balboa/climate.py

```diff
@@ -0,0 +1,207 @@
+"""Support for Balboa Spa Wifi adaptor."""
+from __future__ import annotations
+
+import math
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    FAN_OFF,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_FAN_MODE,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import (
+    ATTR_TEMPERATURE,
+    PRECISION_HALVES,
+    PRECISION_WHOLE,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+)
+
+from .const import CLIMATE, CLIMATE_SUPPORTED_FANSTATES, CLIMATE_SUPPORTED_MODES, DOMAIN
+from .entity import BalboaEntity
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the spa climate device."""
+    async_add_entities(
+        [
+            BalboaSpaClimate(
+                hass,
+                entry,
+                hass.data[DOMAIN][entry.entry_id],
+                CLIMATE,
+            )
+        ],
+    )
+
+
+class BalboaSpaClimate(BalboaEntity, ClimateEntity):
+    """Representation of a Balboa Spa Climate device."""
+
+    _attr_icon = "mdi:hot-tub"
+    _attr_fan_modes = CLIMATE_SUPPORTED_FANSTATES
+
+    def __init__(self, hass, entry, client, devtype, num=None):
+        """Initialize the climate entity."""
+        super().__init__(hass, entry, client, devtype, num)
+
+        # set the supported features
+        self._attr_supported_features = SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE
+        if self._client.have_blower():
+            self._attr_supported_features |= SUPPORT_FAN_MODE
+
+        # map the blower and heatmodes back and forth
+        self.balboa_to_ha_blower_map = {
+            self._client.BLOWER_OFF: FAN_OFF,
+            self._client.BLOWER_LOW: FAN_LOW,
+            self._client.BLOWER_MEDIUM: FAN_MEDIUM,
+            self._client.BLOWER_HIGH: FAN_HIGH,
+        }
+        self.ha_to_balboa_blower_map = {
+            value: key for key, value in self.balboa_to_ha_blower_map.items()
+        }
+
+        self.balboa_to_ha_heatmode_map = {
+            self._client.HEATMODE_READY: HVAC_MODE_HEAT,
+            self._client.HEATMODE_RNR: HVAC_MODE_AUTO,
+            self._client.HEATMODE_REST: HVAC_MODE_OFF,
+        }
+
+    @property
+    def hvac_modes(self) -> list[str]:
+        """Return the list of supported HVAC modes."""
+        if self._client.get_heatmode() == self._client.HEATMODE_RNR:
+            return [*CLIMATE_SUPPORTED_MODES, HVAC_MODE_AUTO]
+        return CLIMATE_SUPPORTED_MODES
+
```

Comment:
```suggestion
    @property
    def hvac_modes(self) -> list[str]:
        """Return the list of supported HVAC modes."""
        if self._client.get_heatmode() == self._client.HEATMODE_RNR:
            return [*CLIMATE_SUPPORTED_MODES, HVAC_MODE_AUTO]
        return CLIMATE_SUPPORTED_MODES

```

Ideally these are set in `__init__` as `self._attr_hvac_modes` instead

Reply:
Same as above, not physically how the device works.